trace_decoder/src/decoding.rs

```diff
@@ -226,573 +44,531 @@ struct TrieDeltaApplicationOutput {
+                v,
+            )
+        })
+    {
+        slots_nibbles.push(slot);
+
+        // If we are writing a zero, then we actually need to perform a delete.
+        match val == &ZERO_STORAGE_SLOT_VAL_RLPED {
+            false => {
+                storage_trie.insert(slot, val.clone()).context(format!(
+                    "at slot {} with value {}",
+                    CustomFmt(U512::from_big_endian(slot.bytes_be().as_slice())),
+                    U512::from_big_endian(val.as_slice())
+                ))?;
+
+                delta_out
+                    .additional_storage_trie_paths_to_not_hash
+                    .entry(ADDRESS)
+                    .or_default()
+                    .push(slot);
+            }
+            true => {
+                if let Ok(Some(remaining_slot_key)) =
+                    delete_node_and_report_remaining_key_if_branch_collapsed(storage_trie, &slot)
+                {
+                    delta_out
+                        .additional_storage_trie_paths_to_not_hash
+                        .entry(ADDRESS)
+                        .or_default()
+                        .push(remaining_slot_key);
                 }
-
-                Self::process_txn_info(
-                    current_idx,
-                    is_initial_payload,
-                    txn_info,
-                    &mut curr_block_tries,
-                    &mut extra_data,
-                    &other_data,
-                )
-                .map_err(|mut e| {
-                    e.txn_idx(txn_idx);
-                    e
-                })
-            })
-            .collect::<TraceParsingResult<Vec<_>>>()
-            .map_err(|mut e| {
-                e.block_num(other_data.b_data.b_meta.block_number);
-                e.block_chain_id(other_data.b_data.b_meta.block_chain_id);
-                e
-            })?;
-
-        if !self.withdrawals.is_empty() {
-            Self::add_withdrawals_to_txns(
-                &mut txn_gen_inputs,
-                &mut curr_block_tries,
-                self.withdrawals,
-            )?;
+            }
         }
+    }
+
+    nodes_used.storage_accesses.push((ADDRESS, slots_nibbles));
+
+    let addr_nibbles = Nibbles::from_h256_be(ADDRESS);
+    delta_out
+        .additional_state_trie_paths_to_not_hash
+        .push(addr_nibbles);
+    let addr_bytes = trie_state
+        .state
+        .get(addr_nibbles)
+        .context(format!("missing account storage trie {:x}", ADDRESS))?;
+    let mut account = account_from_rlped_bytes(addr_bytes)?;
+
+    account.storage_root = storage_trie.hash();
+
+    let updated_account_bytes = rlp::encode(&account);
+    trie_state
+        .state
+        .insert(addr_nibbles, updated_account_bytes.to_vec())
+        .context(format!(
+            "at slot {}",
+            CustomFmt(U512::from_big_endian(addr_nibbles.bytes_be().as_slice()))
```

Comment:
This conversion is used in multiple places, could be probably moved to the conversion function

Reply:
I agree, but I'd prefer to wait for a future PR to keep the changes low - I think it's risky to bundle too many logic changes with indentation changes :S